### ISSUE TITLE
[question] How does it compare to Prometheus?

### ISSUE THREAD

**Issue by drscre** [facebookincubator/beringei#14]

**Comment by scottfranklin** [comment]
The first big difference is scope. Prometheus is a full-featured monitoring solution that tracks metadata and runs complex queries and detectors, while Beringei is just a data store with put and get methods.

The other big difference is that Beringei stores everything in memory. Data is persisted to disk to allow the service to restart with minimal data loss, but Beringei requires enough memory to store the complete data set.

The reason for this is performance.
We're hoping to publish some real benchamarks against other systems soon (everyone's been out for the holidays), but to speak generally, Beringei is the component that allows Facebook's monitoring system to track billions of metrics and execute queries across a million of them at a time.

Berengei pays the cost of RAM instead of Flash, but in return, it can serve a query for 1M timeseries in a few seconds, which I do not believe Prometheus is able to do (though again, we'll run some benchmarks).

We find this to be a worthwhile tradeoff.
Given that > 90% of queries are for data < 1 day old, we actually reap most of the benefits by only storing 1 day of data in Beringei; the rest is kept in slower, long-term storage ([very, very old slide deck](http://cdn.oreillystatic.com/en/assets/1/event/85/Facebook%E2%80%99s%20Large%20Scale%20Monitoring%20System%20Built%20on%20HBase%20Presentation.pdf)).

To summarize, Beringei is pretty much a component that can be bolted into an existing monitoring ecosystem (such as Prometheus) in order to improve the performance of queries for recent data.

**Comment by drscre** [comment]
I guess, if you give Prometheus enough memory to keep 1 day of metric data, it will cache all this metric data in memory and behave similar to Beringei. 

It would be interesting to see benchmark results.

**Comment by fengyuncrawl** [comment]
@scottfranklin I think the Prometheus can't compare with Beringei,Beringei only support full memory ,it's important value is that the timestamp encode and value encode ,because bit encoding is most save  storage, Frankly speaking,I don‘t care metrics query performance ,its not database!

**Comment by mlgood** [comment]
Are there any remaining questions here?

**Comment by asafm** [comment]
Nope

**Comment by drscre** [comment]
@mlgood 
Actually, yes.
If you give Prometheus enough memory to fit all your 1 day data, it will act very similar to in-memory database.
It makes sence to compare Prometheus and Beringei in setup where Prometheus is allowed to use the same amount of memory as Beringei.
